### DISCUSSION TITLE
Ableton Push 2 with Midi2LR

### DISCUSSION THREAD

**Discussion by Kobbleton** [rsjaffe/MIDI2LR#690] [Q&A]
Hello everybody and have a wonderful day ☀️

Does anybody have an experience with **Ableton Push** and **Midi2LR**?
 
I'm trying to use the knobs but they can only have a value either of 1 or 127. So it would do -max of exposure or +max for example.
I've googled but couldn't find any info about this. 

Would be very grateful for any help 🙏🏻

**Comment by antzpantz** [comment]
I don't have an Ableton Push but you may need to configure your controller
with the Ableton program:

https://help.ableton.com/hc/en-us/articles/209071249-Push-1-2-User-Mode-for-custom-MIDI-mappings

Also this may help with configuration:

https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup

**Reply by Kobbleton** [reply]
Thanks! It was very helpful. Indeed I've missed that you can change the setting of the knob on right-click in Midi2LR.
Now it works as it should. 
Really appreciate 🙏🏻
Now I can make more use of my controller 😁

**Comment by rsjaffe** [comment]
Read https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup#controllers-that-send-relative-values-or-have-unusual-ranges . Your controller sends _twos complement_ values.

**Reply by Kobbleton** [reply]
Thanks! Indeed that was the case. Now it works as it should ✅

**Comment by Kobbleton** [comment]
**For those who are also searching for how to use Ableton Push 2 with Midi2LR, it's possible:**

1. You'll have 11 knobs, and a whole bunch of buttons.
2. All buttons (except the 8x8 pad grid) send Control Change messages. This means they send values of 0 and 127 and can't be used to "toggle" parameters.
3. 8x8 pad grid can be used to "toggle" parameters.
4. The Touchstrip can be mapped but will continue to send Pitch Bend information simultaneously. (It means that it will not be usable in our case. Because it returns a middle value every time you leave it. For example, if you change the exposure by pitch bend when you release the finger it resets to 0). I wish it could be altered to modulation wheel instead, but I didn't find how to do this.
5. LCD and LEDs would be unavailable, so it stays completely unresponsive (but works).

**So in order to use it:**

1. Turn Push on, run Ableton.
2. In MIDI preferences change the Push setting like in the official manual [here](https://help.ableton.com/hc/en-us/articles/209071249-Push-1-2-User-Mode-for-custom-MIDI-mappingsl). Close the Ableton.
3.  Make sure that Ableton is not running. (You should see the Push logo on the screen)
4. Open Lightroom and Midi2LR. 
5. To map the knob - turn it simultaneously and look to the Midi2LR interface. When you touch the knob it sends "Note" value and if you turn it the "CC" value so it could be tricky at first. You need "CC" value to use the knob.
6. When you've got it right-click on the LR command box of that knob and choose "Two's Complement". Further explanation [here](https://github.com/rsjaffe/MIDI2LR/wiki/MIDI-Controller-Setup#controllers-that-send-relative-values-or-have-unusual-ranges)

Also even if **it works**, **I found this controller quite basic**. As knobs are the most valuable in my opinion in Midi2LR workflow, I wish we could have interchangeable pages of knobs, to have multiple of parameters mapped and not only 11.

Absolutely the best controller to use with Midi2LR, in my experience is NI MASCHINE mk3. It has endless capabilities of mapping due to amazing software. You can have multiple pages of knobs, LCD is working perfectly and you can label text values of what each knob is doing to the screen, so no need to remember it. You can map everything to it, so there is almost no need to touch the mouse or keyboard ever again during the editing sessions.

I will do an in-depth review of the workflow with it and share my mappings as soon as I'll have an opportunity to come back home to Ukraine.

If there are any questions left, I'll be happy to help. Have a wonderful day ☀️
